File: aoj/volume100/csharp/A10015.cs
using System;
using System.Collections.Generic;
using System.Linq;

class AOJ
{
	static void Main()
	{
		var kinds    = "SHCD".ToCharArray();
		var indexes  = Enumerable.Range(1, 13);

		var cards = new HashSet<string>();
		foreach (var kind in kinds)
			foreach (var index in indexes)
				cards.Add(ToCards(kind, index));

		var noCards = int.Parse(Console.ReadLine());
		for (var ci = 0; ci < noCards; ci++)
		{
			var card = Console.ReadLine();
			cards.Remove(card);
		}

		foreach (var kind in kinds)
			foreach (var index in indexes)
				if (cards.Contains(ToCards(kind, index)))
					Console.WriteLine(ToCards(kind, index));
	}

	static string ToCards(char kind, int index)
	{
		return kind + " " + index;
	}
}

File: aoj/volume100/php/A10015.php
<?php
$kinds = ["S", "H", "C", "D"];
$indexes = range(1, 13);

$cards = [];
foreach ($kinds as $kind)
    foreach ($indexes as $index)
        $cards[$kind . $index] = true;

unset($kind, $index);

fscanf(STDIN, "%d", $nocards);
for ($i = 0; $i < $nocards; $i++) {
    fscanf(STDIN, "%s %d", $kind, $index);
    $cards[$kind . $index] = false;
}

foreach ($kinds as $kind)
    foreach ($indexes as $index)
        if ($cards[$kind . $index])
            printf("%s %d\n", $kind, $index);
